docs/developer/contributing/development-accessibility-tests.asciidoc

```diff
@@ -1,6 +1,52 @@
 [[development-accessibility-tests]]
 == Automated Accessibility Testing
 
+=== Writing Tests
+
+To write an accessibility test, use the provided accessibility service `getService('a11y')`. Accessibility tests are fairly straightforward to write as https://github.com/dequelabs/axe-core[axe] does most of the heavy lifting. Navigate to the UI that you need to test, then call `testAppSnapshot();` from the service imported earlier to make sure axe finds no failures. Navigate through every portion of the UI for the best coverage.
+
+An example test might look like this:
+["source","js"]
+----
+export default function ({ getService, getPageObjects }) {
+  const { common, home } = getPageObjects(['common', 'home']);
+  const a11y = getService('a11y'); /* this is the wrapping service around axe */
+
+  describe('Kibana Home', () => {
+    before(async () => {
+      await common.navigateToApp('home'); /* navigates to the page we want to test */
+    });
+
+    it('Kibana Home view', async () => {
+      await a11y.testAppSnapshot(); /* this expects that there are no failures found by axe */
```

Comment:
I think we should also always have an assertion against an element on the page to verify that we're where we expect to be. Otherwise the a11y snapshot might pass, but only because it's checking the wrong UI! Am I overthinking this?

Reply:
I can but didn't to avoid cluttering it up.

That type of thing is common in all types of tests and often in the accessibility tests we need to find a button or link or something to move on to the next screen for the test which is effectively doing the same thing (leaving just the last test "vulnerable" to this false-positive).

That all said, do you still think it's worth adding? I don't mind doing it if you still think it's helpful just wanted to share why I didn't.

Reply:
> often in the accessibility tests we need to find a button or link or something to move on to the next screen for the test which is effectively doing the same thing

How would the developer be made aware of this button or link click failing to trigger navigation?

> That all said, do you still think it's worth adding? I don't mind doing it if you still think it's helpful just wanted to share why I didn't.

🤷  Maybe worth a comment or mention.

Reply:
> How would the developer be made aware of this button or link click failing to trigger navigation?

The test crashes when it can't find the button

Reply:
I suppose the button could be present, but clicking it could be a no-op?

Reply:
Added a comment about making sure you're on the right page